skimage/transform/_hough_transform.pyx

```diff
@@ -146,7 +146,12 @@ def _hough_ellipse(cnp.ndarray img, int threshold=4, double accuracy=1,
     if img.ndim != 2:
             raise ValueError('The input image must be 2D.')
 
-    cdef Py_ssize_t[:, ::1] pixels = np.row_stack(np.nonzero(img))
+    cdef cnp.ndarray[ndim=2, dtype=cnp.intp_t] noncontiguous_mem_pixels = np.row_stack(np.nonzero(img))
```

Comment:
Why isn't this just `if pixels.shape[1] == 0: return None`? 

You should also add a test in `skimage/transform/test_hough_transform.py`.

Reply:
I created a new array because it is the type of the variable we want to create that causes the error when the image has no non-zero pixel. (`ValueError: Buffer and memoryview are not contiguous in the same dimension.`)

However, I think that the choice of using a contiguous array for a faster processing of all the calculations below this line should not be changed.

The only thing I could come up with was creating a non-contiguous array, testing for its shape, and transform this array into a continuous array if non-zero pixels are present.

I'm also going to add the test. As a side note, this is my first contribution and the first time I work with Cython :)

Reply:
@Eskapp Well, welcome! =) And thank you for this! I hope it won't be the last! 

Ah! I see now why. It still seems clumsy to me. I suggest the following:

```python
if not np.any(img):
    return None
```

And then create `pixels` afterwards. What do you think?